### DISCUSSION TITLE
unexpected token problem

### DISCUSSION THREAD

**Discussion by mr-linch** [alecthomas/participle#258] [Q&A] [answered]
Hey, I need to parse following expression: 

`[$ string $a|b|c]`

in the following struct:

```go
type Expr struct {
	Separtor string   `parser:"'[$' @Ident '$'"`
	Values   []string `parser:"@Ident ('|' @Ident)+ ']'"`
}
```

But when I do it, I receive error `1:1: unexpected token "["`

```go
parser, err := participle.Build[Expr]()
if err != nil {
  log.Fatal(err)
}

const text = "[$ string $a|b|c]"

ast, err := parser.ParseString("", text)
if err != nil {
  log.Fatal(err)
}
```

What is wrong?

**Comment by alecthomas** [comment, marked as answer]
The problem is that `[` and `$` are separate tokens but when you have `'[$'` you're asking the parser to match a single token with those two characters. You need to change it to `'[' '$'`.
